applications/serial_lte_modem/src/slm_at_udp_proxy.c

```diff
@@ -330,15 +328,36 @@ static void udp_thread_func(void *p1, void *p2, void *p3)
 {
 	int ret;
 	int size = sizeof(struct sockaddr_in);
+	struct pollfd fds;
 	char data[NET_IPV4_MTU];
 
 	ARG_UNUSED(p1);
 	ARG_UNUSED(p2);
 	ARG_UNUSED(p3);
 
+	fds.fd = udp_sock;
+	fds.events = POLLIN;
+
 	do {
-		ret = recvfrom(udp_sock, data, NET_IPV4_MTU, 0,
-			(struct sockaddr *)&remote, &size);
+		ret = poll(&fds, 1, MSEC_PER_SEC * CONFIG_SLM_UDP_POLL_TIME);
+		if (ret < 0) {  /* IO error */
+			LOG_WRN("poll() error: %d", ret);
+			continue;
+		}
+		if (ret == 0) {  /* timeout */
+			continue;
+		}
+		LOG_DBG("Poll events 0x%08x", fds.revents);
```

Comment:
I'd check for POLLERR as well, I recall it can also show up as a poll result from bsdlib. Not sure about POLLHUP for UDP socket, but I guess it would be no hard to verify it either.

Reply:
POLLERR is now also handled in TCP/UDP polling.